encodings/fastlanes/src/for/compute/compare.rs

```diff
@@ -55,7 +56,10 @@ where
 
     // Wrap up the RHS into a scalar and cast to the encoded DType (this will be the equivalent
     // unsigned integer type).
-    let rhs = Scalar::from(rhs).reinterpret_cast(T::PTYPE.to_unsigned());
+    let rhs = Scalar::from(rhs);
+    let rhs = rhs
+        .cast(&rhs.dtype().with_nullability(nullability))?
+        .reinterpret_cast(T::PTYPE.to_unsigned());
```

Comment:
```suggestion
        let rhs = Scalar::primitive(rhs, nullability).reinterpret_cast(T::PTYPE.to_unsigned());
```
I think what you want is this